cmd/image-builder/upload.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/cheggaaa/pb/v3"
+	"github.com/spf13/cobra"
+
+	"github.com/osbuild/bootc-image-builder/bib/pkg/progress"
+	"github.com/osbuild/images/pkg/cloud"
+	"github.com/osbuild/images/pkg/cloud/awscloud"
+)
+
+var ErrNoUploadConfig = fmt.Errorf("missing or incomplete upload configuration")
```

Comment:
```suggestion
var ErrNoUploadConfig = errors.New("missing or incomplete upload configuration")
```

Reply:
I would actually suggest re-thinking this a bit. This error message is not very helpful when a user misses one of the required upload args.

Reply:
I did this now. It will give nicer errors - but the downside is that we get the nice errors only in the "image-builder upload" case, when "build" the upload is implicit so any missing parameter will just silently not upload. This is okay (if unexpected) as long as we have "image-builder upload" to fix it after-the-fact.